### PR TITLE
[PAR-4146] config error wrapping

### PR DESCRIPTION
**Pull request by billhefty**
# Description

- Add additional condition checks to avoid unhandled exceptions
- Updated `eslintrc` to be more specific to the use cases for this repo
  - `browser` and `amd` env to allow for `require` and `define` and `window` objects
  - Ignore unused variables prefixed with `_` - handy to keep them defined so any custom integration can easily make use of the params

## Ticket

List your ticket here

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have deployed and validated my code on a sandbox
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: view/frontend/web/js/view/checkout/summary/shipping-protection-offer.js [hunk 1]
```diff
@@ -3,116 +3,115 @@
...
+  'extendSdk',
+  'ExtendMagento',
+], function (Component, ko, customerData, magentoQuote, getTotalsAction, Extend, ExtendMagento) {
+  'use strict'
+  return Component.extend({
+    defaults: {
+      template: 'Extend_Integration/checkout/summary/shipping-protection-offer',
+    },
+    shouldRenderSP: function () {
+      if (window.ExtendConfig && window.ExtendConfig.environment && window.ExtendConfig.storeId)
```
Comment (billhefty) [inline view/frontend/web/js/view/checkout/summary/shipping-protection-offer.js:21]: Adding `window.ExtendConfig && ` is the only actual change on this line.

File: view/frontend/web/js/view/checkout/summary/shipping-protection-offer.js [hunk 2]
```diff
@@ -3,116 +3,115 @@
...
+        // Swallow error to avoid impacting customer checkout experience
+        /* eslint-disable-next-line no-console */
+        console.error(error)
+      }
+    },
+    initialize: function () {
+      this._super()
+
+      try {
+        Extend.config({
```
Comment (billhefty) [inline view/frontend/web/js/view/checkout/summary/shipping-protection-offer.js:99]: Moved this within the `try` block

**Review by billhefty — COMMENTED** [review]
Left comments regarding actual code changes. The rest are from linting the file for the first time.

File: .eslintrc.js [hunk 3]
```diff
@@ -2,6 +2,8 @@ module.exports = {
   env: {
     es2020: true,
     node: true,
+    browser: true,
```
Comment (milesmcleod) [inline .eslintrc.js:5]: nice, i did not know about this.

**Review by milesmcleod — APPROVED** [review]

**Review by joshuathompsonextend — APPROVED** [review]

**Pull request merged**